lib/developer_portal/config/routes.rb

```diff
@@ -65,7 +65,12 @@
 
       resource :personal_details, :only => [:show, :update]
       resource :password, :only => [:new, :show, :create, :update]
-      resources :invoices, :only => [:index, :show]
+      resources :invoices, :only => [:index, :show] do
+        member do
+          get :payment
+          post :payment_succeeded
```

Comment:
It might not succeed so the naming is misleading

Reply:
I had doubts myself and we can totally rename it just in case the behaviour changes in the future. Right now though the JS part won't call this action unless it was a success.